testdata/xml-validate/project.json

```diff
@@ -0,0 +1,140 @@
+    ],
+    "lists": [
+      {
+        "name": "onlyList",
+        "labels": {
+          "en": "List"
+        },
+        "comments": {
+          "en": "This is the only list"
+        },
+        "nodes": [
+          {
+            "name": "n1",
+            "labels": {
+              "en": "Node 1"
+            },
+            "nodes": [
+              {
+                "name": "n1.1",
+                "labels": {
+                  "en": "Node 1.1"
+                },
+                "nodes": [
+                  {
+                    "name": "n1.1.1",
+                    "labels": {
+                      "en": "Node 1.1.1"
+                    }
+                  }
+                ]
+              }
+            ]
+          }
+        ]
+      }
+    ],
+    "ontologies": [
+      {
+        "name": "onto",
+        "label": "Ontology",
+        "properties": [
+          {
+            "name": "hasSimpleText",
+            "super": [
+              "hasValue"
+            ],
+            "object": "TextValue",
+            "labels": {
+              "en": "Simple Text"
+            },
+            "gui_element": "SimpleText"
+          },
+          {
+            "name": "listProp",
+            "super": [
+              "hasValue"
+            ],
+            "object": "ListValue",
+            "labels": {
+              "en": "List Property"
+            },
+            "gui_element": "List",
+            "gui_attributes": {
+              "hlist": "onlyList"
+            }
+          },
+          {
+            "name": "linkProp",
+            "super": [
+              "hasLinkTo"
+            ],
+            "object": ":CardOneResource",
+            "labels": {
+              "en": "Link to CardOneResource"
+            },
+            "gui_element": "Searchbox"
+          }
+        ],
+        "resources": [
+          {
+            "name": "NormalClass",
+            "super": "Resource",
+            "labels": {
+              "en": "Superclass Resource"
+            },
+            "cardinalities": [
+              {
+                "propname": ":hasSimpleText",
+                "cardinality": "0-n",
+                "gui_order": 0
+              },
+              {
+                "propname": ":listProp",
+                "cardinality": "0-n",
+                "gui_order": 1
+              },
+              {
+                "propname": ":linkProp",
+                "cardinality": "0-n",
+                "gui_order": 2
+              }
+            ]
+          },
+          {
+            "name": "CardOneResource",
+            "super": "Resource",
+            "labels": {
+              "en": "Resource with One Cardinality"
+            },
+            "cardinalities": [
+              {
+                "propname": ":hasSimpleText",
+                "cardinality": "1",
+                "gui_order": 0
+              }
+            ]
+          },
+          {
+            "name": "SubClass",
```

Comment:
Regarding the inheritance I added no cardinalities, but in the data a resource of this class has the inherited cardinality. Is this not what you meant @BalduinLandolt ?

Reply:
Well having subclasses that inherit the cardinality of the superclass is quite common, so that needs to be included. I can have the sub-class with a correct cardinality, so that again we have an example that should not appear and then a cardinality violation of the superclass. What do you think?

```
    <!-- This property has a cardinality 1 -->
    <resource label="Max Cardinality Violation" restype=":CardOneResource" id="max-card-violation">
        <text-prop name=":hasSimpleText">
            <text encoding="utf8">Text Value One</text>
            <text encoding="utf8">Text Value Two</text>
        </text-prop>
    </resource>

    
    <!-- This cardinality restriction is inherited from the superclass. 
           The example is correct and should not appear on the error report. 
           -->
    <resource label="Max Cardinality Violation" restype=":SubClass" id="inherited-cardinality-correct">
        <text-prop name=":hasSimpleText">
            <text encoding="utf8">Text Value One</text>
        </text-prop>
    </resource>
```

Reply:
It is a good point that it bloats the data. To ensure that "complex" things will pass, I will make a file that is correct and should never fail. You are right that a mixture is not good.